### PR TITLE
Add a cluster setting to disallow loading fielddata on _id field

### LINKED ISSUE #43599: Prevent fielddata loading for _id
**Describe the feature**: Need a way to prevent fielddata memory issues due to aggregating / sorting on `_id`

### PR DESCRIPTION
**Pull request by jimczi**
This change adds a dynamic cluster setting named `indices.id_field_data.enabled`.
When set to `false` any attempt to load the fielddata for the `_id` field will fail
with an exception. The default value in this change is set to `false` in order to prevent
fielddata usage on this field for future versions but it will be set to `true` when backporting
to 7x. When the setting is set to true (manually or by default in 7x) the loading will also issue
a deprecation warning since we want to disallow fielddata entirely when https://github.com/elastic/elasticsearch/issues/26472 is implemented.

Closes #43599

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java [hunk 1]
```diff
@@ -168,6 +169,9 @@
     public static final String INDICES_SHARDS_CLOSED_TIMEOUT = "indices.shards_closed_timeout";
     public static final Setting<TimeValue> INDICES_CACHE_CLEAN_INTERVAL_SETTING =
         Setting.positiveTimeSetting("indices.cache.cleanup_interval", TimeValue.timeValueMinutes(1), Property.NodeScope);
+    public static final Setting<Boolean> INDICES_ID_FIELD_DATA_ENABLED_SETTING =
+        Setting.boolSetting("indices.id_field_data.enabled", false, Property.Dynamic, Property.NodeScope);
```
Comment (jpountz) [inline server/src/main/java/org/elasticsearch/indices/IndicesService.java]: if I understand correctly, the default will become true upon backport?

**Review by jpountz — APPROVED** [review]
Thanks for fixing this longstanding issue!

File: server/src/main/java/org/elasticsearch/indices/IndicesService.java (hunk 1 above)
Comment (jimczi) [reply server/src/main/java/org/elasticsearch/indices/IndicesService.java]: Yes that's the plan

**Comment by thekofimensah** [comment]
This is amazing! I can really use this.. I have users that accidentally try to aggregate on _id, and it destroys my memory.. 1.5gb used until I clear the cache for it.

**Review by jpountz — APPROVED** [review]

**Comment by eedugon** [comment]
@jimczi , shouldn't we add the new setting in this document as a reference? https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-id-field.html

It's the doc where we recommend against sorting or aggregating towards `_id`:

>The value of the _id field is also accessible in aggregations or for sorting, but doing so is discouraged as it requires to load a lot of data in memory.

**Comment by nezda** [comment]
Would be awesome to backport this to 6.8

**Comment by jasontedor** [comment]
Sorry @nezda but we won’t backport this to 6.8 because our policy is that we don’t add new features to versions in maintenance mode.

**Comment by sachin-frayne** [comment]
There are no reference docs on our site about this setting

**Pull request merged**